src/arrayMethodJoin.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let separatorValue = separator;
+
+    if (separatorValue === undefined) {
+      separatorValue = ',';
+    }
+
+    const arr = this;
+    let result = '';
+
+    if (arr.length === 0) {
+      return '';
+    }
+
+    for (const element of arr) {
```

Comment:
If you need an index regular for loop with `let i = 0; i < this.length; i++` would be better